stonesoup/plotter.py

```diff
@@ -304,3 +309,115 @@ def create_artists(self, legend, orig_handle,
         self.update_prop(p, orig_handle, legend)
         p.set_transform(trans)
         return [p]
+
+
+class TimeBasedPlotter(Base):
+
+    plotting_data = Property(list)
+    legend_key = Property(str, default='Not specified', doc="Todo")
+    plotting_keyword_arguments = Property(dict, default=None, doc='Todo')
+
+    def __init__(self, *args, **kwargs):
+        class_keywords, plotting_keywords = self.get_plotting_keywords(kwargs)
+        super().__init__(*args, **class_keywords)
+        self.plotting_data = copy.copy(self.prepare_data(self.plotting_data))
+        self.plotting_keyword_arguments = plotting_keywords
+
+    def get_plotting_keywords(self, kwargs):
+        """Splits keyword arguments needed for this class. Other keyword arguments are used in the
+        matplotlib.pyplot.plot function
+
+        Parameters
+        ----------
+        kwargs : dict
+            Keyword arguments for this class and additional arguments to be passed to plot function
+
+        Returns
+        -------
+        : :class:`dict`
+            keyword arguments to be used by the class
+        : :class:`dict`
+            keyword arguments to be the matplotlib.pyplot.plot function
+        """
+        plotting_keywords = {}
+        class_keywords = {}
+        for key, value in kwargs.items():
+            if key in self._properties.keys():
+                class_keywords[key] = value
+            else:
+                plotting_keywords[key] = value
+        return class_keywords, plotting_keywords
+
+    @staticmethod
```

Comment:
I think the static methods need some documentation for them to appear in the docs - it would be particularly useful for the `run_animation` function.